cmd/build.go

```diff
@@ -155,19 +155,21 @@ func build(config *buildCommandConfig) error {
 		return errors.New("Cannot specify --push or --push-url without a --tag")
 	}
 
-	// check to see if the pushURL provided is in the expected format
-	// Valid examples would be:
-	// https://www.appsody.co.uk
-	// http://www.appsody.co.uk
-	// www.appsody.co.uk
-	// appsody.co.uk
-	matched, err := regexp.MatchString("^(http:\\/\\/www\\.|https:\\/\\/www\\.|http:\\/\\/|https:\\/\\/)?[a-z0-9]+([\\-\\.]{1}[a-z0-9]+)*\\.[a-z]{2,5}(:[0-9]{1,5})?(\\/.*)?$", config.pushURL)
-	if err != nil {
-		return err
-	}
+	if config.pushURL != "" {
+		// check to see if the pushURL provided is in the expected format
+		// Valid examples would be:
+		// https://www.appsody.co.uk
+		// http://www.appsody.co.uk
+		// www.appsody.co.uk
+		// appsody.co.uk
+		matched, err := regexp.MatchString("^(http:\\/\\/www\\.|https:\\/\\/www\\.|http:\\/\\/|https:\\/\\/)?[a-z0-9]+([\\-\\.]{1}[a-z0-9]+)*\\.[a-z]{2,5}(:[0-9]{1,5})?(\\/.*)?$", config.pushURL)
+		if err != nil {
+			return err
+		}
```

Comment:
We would need to put this function in utils, and write specific tests for it.I tested it against `default-route-openshift-image-registry.apps-crc.testing` which should match but it didn't. I think maybe we can put this change in a different PR and merge the build tests for now?

Reply:
I was thinking about this yeah, but wanted to keep it separate from the `build_test` stuff.  I'll extrapolate it into a function and write some test cases for it.